src/scheduler/MpiWorld.cpp

```diff
@@ -341,6 +354,53 @@ std::string MpiWorld::getHostForRank(int rank)
     return host;
 }
 
+std::vector<int> MpiWorld::getLocalRanks()
+{
+    return getRanksForHost(thisHost);
+}
+
+std::vector<int> MpiWorld::getRanksForHost(const std::string& host)
+{
+    assert(rankHosts.size() == size);
+
+    std::vector<int> ranksForHost;
+    for (int i = 0; i < rankHosts.size(); i++) {
+        if (rankHosts.at(i) == host) {
+            ranksForHost.push_back(i);
+        }
+    }
+
+    return ranksForHost;
+}
+
+int MpiWorld::getLocalMaster()
+{
+    return getMasterForHost(thisHost);
+}
+
+std::vector<int> MpiWorld::getRemoteMasters()
+{
+    std::set<std::string> targetHosts(rankHosts.begin(), rankHosts.end());
+    targetHosts.erase(thisHost);
+
+    std::vector<int> remoteMasters;
+    for (const std::string& host : targetHosts) {
+        remoteMasters.push_back(getMasterForHost(host));
+    }
+
+    return remoteMasters;
+}
+
+// The local master for an MPI world is defined as the lowest rank assigned to
+// this host
+int MpiWorld::getMasterForHost(const std::string& host)
+{
+    std::vector<int> ranks = getRanksForHost(host);
+    assert(!ranks.empty());
```

Comment:
We only call this method internally, with values from the host list that we have been broadcasted, and that necessarily must match to a rank (otherwise it wouldn't have been sent in the first place).
Thus why we `assert` here insted of throwing an exception.